### PR TITLE
Feature/dot 115 setup a company overview page

### PR DESCRIPTION
**Pull request by Terbau**
Closes: #346 

## Changelog

- Adds a company page for each company at `/company/[id]`
- Adds a temporary company list at `/company`
- Renamed route of the old company page to `/company-info` as well as renamed stuff related to this internally
- Adds `next-superjson-plugin` as a dependency for apps/web for seamless json serialization in the pages directory just like trpc serialization. 

Please note that the event list design on the company page is temporary. Functionality works for fetching events but not for fetching job listings.

## Preview
Screenshot from 2023-04-13 16-49-00

### CODE REVIEW

File: apps/web/src/components/views/CompanyInfoView/index.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import OurProducts from "./OurProducts"
+import { Title } from "@dotkomonline/ui"
+
+export type Content = { content: PortableTextProps["blocks"] }
+
+interface CompanyInfoViewProps {
+  companyInfoContent: Content[]
+}
+export const CompanyInfoView: FC<CompanyInfoViewProps> = (props: CompanyInfoViewProps) => {
+  const [header, interest, product, info, contact] = props.companyInfoContent
```
Comment (matsjla) [inline apps/web/src/components/views/CompanyInfoView/index.tsx:16]: Maybe make it an object instead? It's harder to keep track of an array index over time :p

File: apps/web/src/components/views/CompanyView/CompanyEventList.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  return (
+    <div className="w-full">
+      <h2 className="border-none">{props.title}</h2>
+      <ul className="mt-4 flex list-none flex-col gap-y-2">
+        {props.isLoading ? (
+          <p>Loading...</p>
+        ) : (
+          (props.events || []).map((event) => (
+            <li key={event.id}>
+              <a
```
Comment (matsjla) [inline apps/web/src/components/views/CompanyView/CompanyEventList.tsx:20]: Should probably be a Next link

File: apps/web/src/components/views/CompanyView/index.tsx [hunk 3]
```diff
@@ -1,50 +1,70 @@
...
+                <Link href={website} target="_blank">
+                  <img src={image} alt="Company logo" className="w-full" />
+                </Link>
+              </div>
+            )}
+
+            <div className="text-blue-12 flex flex-col gap-y-2 px-1 text-lg [&>*]:flex [&>*]:items-center [&>*]:gap-x-2">
+              <div>
+                <Icon icon="ph:globe-bold" width="28"></Icon>
+                <a className="text-blue-11 hover:text-blue-10" href={website}>
```
Comment (kharann) [inline apps/web/src/components/views/CompanyView/index.tsx:37]:
```
const icons = [
{
    icon: "material-symbols:phone-enabled",
    text: "1234151",
    href: "tel:1234151"
},
....
] as const

icons.map( icon => ....)
```

**Review by kharann — CHANGES_REQUESTED** [review]

File: apps/web/src/components/views/CompanyView/index.tsx (hunk 3 above)
Comment (Terbau) [reply apps/web/src/components/views/CompanyView/index.tsx]: Fixed :)

File: apps/web/src/components/views/CompanyInfoView/index.tsx (hunk 1 above)
Comment (Terbau) [reply apps/web/src/components/views/CompanyInfoView/index.tsx:16]: Resolving as its unrelated to this PR.

File: apps/web/src/components/views/CompanyView/CompanyEventList.tsx (hunk 2 above)
Comment (Terbau) [reply apps/web/src/components/views/CompanyView/CompanyEventList.tsx]: Resolving as its unrelated to this PR.

File: apps/web/src/components/views/CompanyView/index.tsx [hunk 4]
```diff
@@ -1,50 +1,68 @@
...
+          <div className="border-blue-7 flex h-fit flex-col gap-y-3 rounded-lg border-none sm:gap-y-2">
+            {image && (
+              <div className="relative mb-4 h-64 w-full overflow-hidden rounded-lg bg-[#fff]">
+                <a href={website} target="_blank" rel="noreferrer">
+                  <Image src={image} alt="Company logo" layout="fill" objectFit="contain" className="w-full" />
+                </a>
+              </div>
+            )}
+
+            <div className="text-blue-12 flex flex-col gap-y-2 px-1 text-lg [&>*]:flex [&>*]:items-center [&>*]:gap-x-2">
```
Comment (kharann) [inline apps/web/src/components/views/CompanyView/index.tsx:41]: Why not put items-center and gap-x-2 on the divs below?

Comment (Terbau) [reply apps/web/src/components/views/CompanyView/index.tsx]: That was some old styling code from a bit earlier when the div contained multiple hardcoded divs. Fixed it now :)

**Pull request merged**